### DISCUSSION TITLE
How do I save a tree

### DISCUSSION THREAD

**Discussion by arjunsrivatsa** [tskit-dev/msprime#1732] [Q&A] [answered]
It is incompatible with pickle.dump, and I cant convert it to a dict, even though it already seems to be in that format

**Comment by arjunsrivatsa** [comment]
This is resolved because I manually made a dictionary, but a pickleable object would be great for the future

**Reply by benjeffery** [reply]
Hi @arjunsrivatsa,
Do you mean a single `Tree` from a `TreeSequence`? Storing such an object as a pickle isn't planned as it can't exist without an underlying tree sequence. `Tree` has the method `as_dict_of_dicts()` which may be of use?

**Comment by jeromekelleher** [comment]
Can you give an example of what you'd like to do with this @arjunsrivatsa? The Tree and TreeSequence classes are bound up in terms of their memory for efficiency, so Trees aren't supported as independent entities. If you can show us some code outlining why you'd like to pickle a Tree we might be able to help.

**Reply by arjunsrivatsa** [reply]
I am using the tree as a portion of a tangential simulation, and as such needed a way for the user to recover it after I had generated data from my program. Random seeding is a possibility, but rerunning the simulation just to recover the trees didn't seem to be the best solution.

**Reply by jeromekelleher** [reply]
I see, thanks @arjunsrivatsa. Is there a particular reason to not save the TreeSequence object then? This is well supported and very efficient - see the [dump](https://tskit.dev/tskit/docs/stable/python-api.html#tskit.TreeSequence.dump) method. Unless you definitely want to save a particular tree in the sequence only?

**Reply by arjunsrivatsa** [reply]
Thanks, that would work

**Comment by jeromekelleher** [comment, marked as answer]
To summarise the conversation above - it's usually better to save the TreeSequence object rather than individuals trees, using the [TreeSequence.dump](https://tskit.dev/tskit/docs/stable/python-api.html#tskit.TreeSequence.dump) method.

**Reply by hyanwong** [reply]
Also if you have a large number of trees and just want to save one, you can use `keep_intervals` to reduce the tree sequence to a single tree and then save. One gotcha with this: unless you `trim()` the resulting tree sequence, you could actually have up to 3 trees because there could be "empty" regions where there are no edges before and/or after the tree of interest, and an empty regions counts as a tree (just an empty one). If you `.trim()` you get rid of these, but potentially change the meaning of the genomic coordinates. But if you only want 1 tree, perhaps these coords are irrelevant.
